unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+
+    def test_unregister(self, redis_client_auto_register):
+        """checks whether the redis.db.publish is called with correct arguments"""
+        data = {"name": AREA_ID, "transaction_id": AREA_ID}
+        self.redis_client = redis_client_auto_register
+        self.redis_client.redis_db.publish = MagicMock()
+        self.redis_client.is_active = True
+        self.redis_client.unregister(is_blocking=False)
+        self.redis_client.redis_db.publish \
+            .assert_called_once_with(f"{AREA_ID}/unregister_participant",
+                                     json.dumps(data))
+
+    def test_unregister_blocking_is_active_false_throws_exception(self, redis_client_auto_register):
+        """checks if is active set to false throws exception"""
+        with pytest.raises(RedisAPIException,
+                           match="API is already unregistered from the market."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = False
+            self.redis_client.unregister(is_blocking=True)
+
+    def test_unregister_blocking_is_active_true_throws_exception(self, redis_client_auto_register):
+        """checks if is active set to true throws exception"""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = True
+            self.redis_client.unregister(is_blocking=True)
+
+    def test_on_register(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_register function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._blocking_command_responses = \
+            {"register": {"transaction_id": f"{TRANSACTION_ID}"}}
+        redis_client_auto_register._on_register(message)
+
+    def test_on_unregister(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_unregister function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID,
+                            "response": "success"}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._blocking_command_responses \
+            = {"unregister": {"transaction_id": f"{TRANSACTION_ID}"}}
+        redis_client_auto_register._on_unregister(message)
```

Comment:
Here https://github.com/gridsingularity/d3a-api-client/pull/163/files#r695724815 applies to, also that here `self.active` should be False